### PR TITLE
Macro documentation in filter engine

### PR DESCRIPTION
**Pull request by fdurand**
# Description
Extra documentation to use the macro in the filter engine

# Impacts
More documentation

# Delete branch after merge
YES

# NEWS file entries

## Enhancements
* Macro documentation for filter engine

### CODE REVIEW

File: docs/installation/advanced_access_configuration.asciidoc [hunk 1]
```diff
@@ -304,6 +304,116 @@ Under certain circumstances, for example to show an AUP after a successful 802.1
...
+
+  PacketFence-UserName = ${join(":",$radius_request.User-Name,"Super")}
+
+will assign to PacketFence-UserName a join of the separator and the values.
+
+ User-Name = "bobey" -> PacketFence-UserName = "bobey:Super"
+
+==== substr ====
+
+This macro will return an substring
```
Comment (nqb) [inline docs/installation/advanced_access_configuration.asciidoc]:
```suggestion
This macro will return a substring
```

File: docs/installation/advanced_access_configuration.asciidoc [hunk 2]
```diff
@@ -304,6 +304,116 @@ Under certain circumstances, for example to show an AUP after a successful 802.1
...
+
+ PacketFence-UserName = ${substr($radius_request.User-Name,0, 5)}
+
+will assign to PacketFence-UserName the substring
+
+ User-Name = "ZammitLudovic" -> PacketFence-UserName = "Zammit"
+
+==== macToEUI48 ====
+
+This macro will return the EUI48 format of a mac address
```
Comment (nqb) [inline docs/installation/advanced_access_configuration.asciidoc]:
```suggestion
This macro will return the EUI48 format of a MAC address
```

File: docs/installation/advanced_access_configuration.asciidoc [hunk 3]
```diff
@@ -304,6 +304,116 @@ Under certain circumstances, for example to show an AUP after a successful 802.1
...
+
+ PacketFence-UserName = ${macToEUI48($radius_request.Calling-Station-Id)}
+
+will assign to PacketFence-UserName the EUI48 mac address
```
Comment (nqb) [inline docs/installation/advanced_access_configuration.asciidoc]:
```suggestion
will assign to PacketFence-UserName the EUI48 MAC address
```

File: docs/installation/advanced_access_configuration.asciidoc [hunk 4]
```diff
@@ -304,6 +304,116 @@ Under certain circumstances, for example to show an AUP after a successful 802.1
...
+
+ Calling-Station-Id = "00:10:7f:38:89:9d" -> PacketFence-UserName = "00-10-7F-38-89-9D"
+
+==== random_from_range ====
+
+This macro will return a random number from a range
+
+ Session-Timeout  = ${random_from_range("10620..12600")}
+
+will assign a number between 10620 and 12600 to Session-Timeout
```
Comment (nqb) [inline docs/installation/advanced_access_configuration.asciidoc]:
```suggestion
will assign a random number between 10620 and 12600 to Session-Timeout
```

File: docs/installation/advanced_access_configuration.asciidoc [hunk 5]
```diff
@@ -304,6 +304,116 @@ Under certain circumstances, for example to show an AUP after a successful 802.1
...
+
+ Session-Timeout = 11343
+
+==== log ====
+
+This macro will log a message in the packetfence.log file
+
+ PacketFence-UserName = ${log($radius_request.User-Name." logged")}
+
+will log the value of the radius request attribute User-Name and logged
```
Comment (nqb) [inline docs/installation/advanced_access_configuration.asciidoc]:
```suggestion
will log the value of the RADIUS request attribute User-Name and logged
```

File: docs/installation/advanced_access_configuration.asciidoc [hunk 6]
```diff
@@ -304,6 +304,116 @@ Under certain circumstances, for example to show an AUP after a successful 802.1
...
+
+ PacketFence-UserName = ${replace($radius_request.User-Name,"z","r")}
+
+will replace the letter z by the letter r in the string
+
+ User-Name = "zabbit" -> PacketFence-UserName = "rabbit"
+
+==== BuildFromMatch ====
+
+This macro will apply a regex on the attribute value to be able able to build anither value
```
Comment (nqb) [inline docs/installation/advanced_access_configuration.asciidoc]:
```suggestion
This macro will apply a regular expression on the attribute value to be able able to build another value
```

**Review by nqb — CHANGES_REQUESTED** [review]
- You don't need to add extra "===" after a title, only before

File: docs/installation/advanced_access_configuration.asciidoc [hunk 7]
```diff
@@ -304,6 +304,116 @@ Under certain circumstances, for example to show an AUP after a successful 802.1
...
+
+  TLS-Stripped-UserName = ${BuildFromMatch($radius_request.TLS-Client-Cert-Common-Name,"^[^@]+","$0")}
+
+will extract the value before the @ sign and assign this value
+
+ TLS-Client-Cert-Common-Name = "zammit@packetfence.org" -> TLS-Stripped-UserName = zammit
```
Comment (nqb) [inline docs/installation/advanced_access_configuration.asciidoc]:
```suggestion
 TLS-Client-Cert-Common-Name = "zammit@packetfence.org" -> TLS-Stripped-UserName = "zammit"
```

**Review by nqb — CHANGES_REQUESTED** [review]

**Comment by fdurand** [comment]
BUMP

**Review by nqb — APPROVED** [review]

**Pull request merged**